### ISSUE TITLE
[FEATURE] Checkpointing

### ISSUE THREAD

**Issue by le1nax** [MouseLand/cellpose#1351]
Hello,

I was confused to see that the default training pipeline does save state_dicts at lowest validation loss? 
Also I am curious why there is no validation splitting? There is "test data" (=validation data?) which you can input but I dont see where it is used to checkpoint? 

Kind Regards
Daniel
